R/findNLDI.R

```diff
@@ -0,0 +1,395 @@
+
+  # Should sf be used? Both no_sf and pkg.env must agree
+  use_sf = all(pkg.env$local_sf, !no_sf)
+
+  #Should the basin be identified?
+  getBasin = ("basin" %in% find)
+
+  # From the collection of all possible origins, pick 1 and remove NULLS
+  starter <- tc(c(
+    list(
+      comid = comid,
+      nwis = nwis,
+      wqp = wqp,
+      huc12 = huc12,
+      location = location
+    ),
+    origin
+  ))
+
+  # a single starting location must be given ...
+  if (is.null(starter) | length(starter) > 1) {
+    stop("Define a single starting point. Use `find` to identify other resources.")
+  }
+
+  # Ensure nav types are valid
+
+  bad.nav = !nav %in% c("UM", "UT", "DD", "DM")
+
+  if (any(bad.nav)) {
+    stop(nav[bad.nav], " not a valid navigation. Use one of: UM, UT, DD, DM")
+  }
+
+  # name of starter
+  start_type = names(starter)
+
+  # If location, then ensure lng is first argument (hack for USA features)
+  if (start_type == 'location') {
+    if (location[1] > 0) {
+      stop("Please provide location in the form c(lng,lat)")
+    }
+
+  # Must convert location to COMID for latter tracing and discovery ...
+    tmp_url = paste0(
+      pkg.env$nldi_base,
+      "comid/position?coords=POINT%28",
+      location[1] ,
+      "%20",
+      location[2] ,
+      "%29"
+    )
+
+    tmp_return = get_nldi(tmp_url, "feature", use_sf = use_sf)
+
+    # Override starter with location based COMID
+    starter = list("comid" = tmp_return$identifier)
+
+  }
+
+  # Reset (if needed)
+  start_type = names(starter)
+
+  # Defining the origin URL.
+    # Make align request with formal name from offerings
+    # If NWIS, add "USGS-" prefix
+  start_url = paste0(
+    valid_ask(pkg.env$current_nldi, type = start_type)$good$feature,
+    "/",
+    ifelse(start_type == "nwis", paste0("USGS-", starter), starter),
+    "/"
+  )
+
+  # Makes sure that all requested featrues to find are valid and name-aligned
+  if (!is.null(find)) {
+    find = valid_ask(pkg.env$current_nldi, type = find)$good$source
+  }
+
+  # Set empty lists to store origin, navigation, featrues, and basin requests ...
+  start <- navigate <- features <- basin <- list()
+
+  # Set origin url
+  start[["start"]] <- start_url
+
+  # Build navigation URLS, doubl check the Nav is valid
+  for (i in seq_along(nav)) {
+    if (nav[i] %in% c("UM", "UT", "DD", "DM")) {
+      navigate[[nav[i]]] = paste0(start_url, "navigate/", nav[i])
+    }
+  }
+
+  # Build basin URL
+  if (getBasin) {
+    basin[["basin"]] = paste0(start_url, "basin")
+  }
+
+  # Build find URLs
+  if (length(find) > 0) {
+    features = lapply(navigate, paste0, paste0("/", find))
+  }
+
+  # Add distance controls to features
+  if (!is.null(distance_km)) {
+    features = lapply(features, paste0, paste0("?distance=", distance_km))
+  }
+
+  # Add distance controls to navigations
+  if (!is.null(distance_km)) {
+    navigate = lapply(navigate, paste0, paste0("?distance=", distance_km))
+  }
```

Comment:
distance_km is required by the `navigation` end point. Need to have a real default. -- I'd suggest 100km.

Reply:
distance_km now defaults to 100